hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleNetworkContext.java

```diff
@@ -368,24 +382,35 @@ public void deserialize(final SerializableDataInputStream in, final int version)
             firstConsTimeOfCurrentBlock = firstBlockTime == null ? null : firstBlockTime.toJava();
             blockNo = in.readLong();
             if (version >= RELEASE_0270_VERSION) {
-                stakingRewardsActivated = in.readBoolean();
-                totalStakedRewardStart = in.readLong();
-                totalStakedStart = in.readLong();
-                pendingRewards = in.readLong();
-                if (version >= RELEASE_0300_VERSION) {
-                    final var expiryCapacityUsed = in.readLong();
-                    final var lastExpiryUsage = readNullable(in, RichInstant::from);
-                    expiryUsageSnapshot =
-                            new DeterministicThrottle.UsageSnapshot(
-                                    expiryCapacityUsed,
-                                    (lastExpiryUsage == null) ? null : lastExpiryUsage.toJava());
-                }
+                readPreBlockData(in, version);
             }
             blockHashes.clear();
             in.readSerializable(true, () -> blockHashes);
         }
     }
 
+    private void readPreBlockData(final SerializableDataInputStream in, final int version)
```

Comment:
nit: method name doesn't really suggest the content logic which includes checks for version 0.30 and 0.31.
Q: Is this supposed to be reusable across upgrades e.g. in future would you want this to do the same in 0.34 to 0.35?

Reply:
You caught me, this was just a hack to reduce the cognitive complexity of the method for Sonar. 😓 

Will ponder a cleaner refactoring. 🤔